app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -144,15 +182,23 @@ public void addNewHabit(Habit habit) {
      */
     public void addHabitToDatabase(String title, String reason, Date date){
         // Handling of adding a habit to firebase
-        final CollectionReference collectionReference = _db.collection("Habits");
         HashMap<String, Object> habitData = new HashMap<>();
 
+        // find lowest positive non zero habitId that is not used by a habit currently
+        long habitIdCounter = 1;
+        while (_habitIdSet.contains(habitIdCounter)) {
+            habitIdCounter++;
+        }
+        _habitIdSet.add(habitIdCounter);
         habitData.put("title", title);
         habitData.put("reason", reason);
         habitData.put("date", date);
+        habitData.put("habitId", habitIdCounter);
+
+        Long habitIdDoc = habitIdCounter;
```

Comment:
What is the point of this variable? I see it's only used once and all we do is call `.toString()` on it.

Reply:
Simply to convert the long to a Long so we can convert the habit ID into a string. I can clean it up to remove the variable if needed.